### ISSUE TITLE
Charge/Intercept speed should always be the same

### ISSUE THREAD

**Issue by Klimpb** [magey/tbc-warrior#36]
Charge/Intercept takes your current speed when casting the spell into consideration to determine the charge/intercept speed itself(slower if slowed, faster if mounted etc), this behavior is incorrect for TBC.

Example: https://streamable.com/u2t9wh

**Comment by magey** [comment]
[Relevant patch note from 2.1.0](https://gist.github.com/magey/6beb165ea93cdb010d5f22a21a9db5c2#file-patch-txt-L4668):

```
  * Charge, Intercept and Intervene now suppress all Slowing effects
    while charging the target.
```

**Comment by badabingcrosby** [comment]
On the TBC Beta right now, and throughout Classic I believe, charges get interrupted if a stun / root is applied to them when a player is charging. 
That's not consistent with how charges worked in TBC, here are some TBC clips : 
1. https://www.youtube.com/watch?v=-41Dgo6auCA&t=144s
2. https://youtu.be/9qv4nbt-NNk?t=941
3. https://www.youtube.com/watch?v=vpEa508cxGE&feature=youtu.be&t=995
Seems like the warrior either gets teleported to the destination when the effect is applied (#1) or at the very least when it expires (#2 and #3).
This behaviour has some balance implications for arena, especially against classes that warriors struggle against already like mages.

**Comment by magey** [comment]
Both of these bugs were fixed in 2.5.1.38521:

Charge/intercept ignoring slows: https://streamable.com/4jwczi

Warrior getting to his charge destination even when interrupted by a stun:


https://user-images.githubusercontent.com/44882404/116684327-83c98c80-a9b9-11eb-84ba-ed0d0c6ee9a3.mp4
